relayer/crates/starknet-test-components/src/impls/bootstrap/deploy_contracts.rs

```diff
@@ -0,0 +1,363 @@
+                &LevelInfo,
+            )
+            .await;
+
+        let ics20_contract_address = {
+            let owner_call_data = cairo_encoding
+                .encode(&ibc_core_address)
+                .map_err(Bootstrap::raise_error)?;
+
+            let erc20_call_data = cairo_encoding
+                .encode(&erc20_class_hash)
+                .map_err(Bootstrap::raise_error)?;
+
+            chain
+                .deploy_contract(
+                    &ics20_class_hash,
+                    false,
+                    &[owner_call_data, erc20_call_data].concat(),
+                )
+                .await
+                .map_err(Bootstrap::raise_error)?
+        };
+
+        logger
+            .log(
+                &format!("deployed ICS20 contract to address: {ics20_contract_address:?}"),
+                &LevelInfo,
+            )
+            .await;
+
+        {
+            let register_client = MsgRegisterClient {
+                client_type: short_string!("07-tendermint"),
+                contract_address: comet_client_address,
+            };
+
+            let calldata = cairo_encoding
+                .encode(&register_client)
+                .map_err(Bootstrap::raise_error)?;
+
+            let call = Call {
+                to: *ibc_core_address,
+                selector: selector!("register_client"),
+                calldata,
+            };
+
+            let message = StarknetMessage::new(call);
+
+            let response = chain
+                .send_message(message)
+                .await
+                .map_err(Bootstrap::raise_error)?;
+
+            logger
+                .log("registered comet client contract with ibc core", &LevelInfo)
+                .await;
+        }
+
+        {
+            // register the ICS20 contract with the IBC core contract
+
+            let register_app = MsgRegisterApp {
+                port_id: PortId::transfer(),
+                contract_address: ics20_contract_address,
+            };
+
+            let register_call_data = cairo_encoding
+                .encode(&register_app)
+                .map_err(Bootstrap::raise_error)?;
+
+            let call = Call {
+                to: *ibc_core_address,
+                selector: selector!("bind_port_id"),
+                calldata: register_call_data,
+            };
+
+            let message = StarknetMessage::new(call);
+
+            let response = chain
+                .send_message(message)
+                .await
+                .map_err(Bootstrap::raise_error)?;
+
+            logger
+                .log("registered ICS20 contract with ibc core", &LevelInfo)
+                .await;
+        }
+
+        chain
+            .ibc_core_contract_address()
+            .set(ibc_core_address)
+            .map_err(|_| {
+                Bootstrap::raise_error("failed to set ibc_core_contract_address on chain")
```

Comment:
Shouldn't the original error be included in the message to avoid losing information?

Reply:
The method only return the original value in `Result`. We are using `OnceLock`, so the value can only be lazily initialized once and then never changed again.

We only perform lazy setting during test, as the chain context needs to be constructed first before deploying the contracts. After that, it won't show up in normal uses.